OpenRA.Mods.Cnc/FileFormats/VqaVideo.cs

```diff
@@ -16,31 +16,32 @@
 
 namespace OpenRA.Mods.Cnc.FileFormats
 {
-	public class VqaReader : IVideo
+	public class VqaVideo : IVideo
 	{
-		public ushort Frames => frames;
-		public byte Framerate => framerate;
-		public ushort Width => width;
-		public ushort Height => height;
-
-		readonly ushort frames;
-		readonly byte framerate;
-		readonly ushort width;
-		readonly ushort height;
-
-		Stream stream;
-		int currentFrame;
-		ushort numColors;
-		ushort blockWidth;
-		ushort blockHeight;
-		byte chunkBufferParts;
-		int2 blocks;
-		uint[] offsets;
-		uint[] palette;
-		uint videoFlags; // if 0x10 is set the video is a 16 bit hq video (ts and later)
-		int sampleRate;
-		int sampleBits;
-		int audioChannels;
+		public ushort FrameCount { get; }
+		public byte Framerate { get; }
+		public ushort Width { get; }
+		public ushort Height { get; }
+
+		public byte[] CurrentFrameData { get; }
+		public int CurrentFrameNumber { get; private set; }
+
+		public bool HasAudio { get; set; }
+		public byte[] AudioData { get; private set; } // audio for this frame: 22050Hz 16bit mono pcm, uncompressed.
```

Comment:
Feels reasonable to promote this to an xml comment.

Reply:
I am not convinced actually - such a comment should generally go on `IVideo`, but this particular comment is specific to the `VqaVideo` implementation of `IVideo`. And we also never expose `VqaVideo` publicly, so if the comment is not on the interface it would still not be visible anywhere.